test/metabase/api/user_test.clj

```diff
@@ -557,6 +558,34 @@
                    (dissoc :user_group_memberships)
                    mt/boolean-ids-and-timestamps)))))))
 
+(deftest updated-user-name-test
+  (testing "Test that `metabase.api.user/updated-user-name` works as intended."
+    (let [names     {:first_name "Test" :last_name "User"}
+          nonames   {:first_name nil :last_name nil}
+          firstname {:first_name "Test" :last_name nil}
+          lastname  {:first_name nil :last_name "User"}]
```

Comment:
Are these two keys always present? If not, add test cases when some of them are missing.

Reply:
They are always present. The function will normally have a full user map passed in. I'm just using straight maps here to keep the test from creating a bunch of temp users. I don't really need the full user anyway in this case.

Reply:
Added a comment in the test to try clarify this a bit.